docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
+
+## Current Challenge
```

Comment:
Would it be good to state the goal of the minimal distribution?

Reply:
Sure, but since I do not have the goal for minimal distribution as well, I will add it as a challenge here.